core/src/main/java/com/linecorp/armeria/common/logging/RequestScopedMdc.java

```diff
@@ -248,7 +253,9 @@ public static void putAll(RequestContext ctx, Map<String, String> map) {
             return;
         }
 
-        synchronized (ctx) {
+        final ReentrantLock reentrantLock = new ReentrantLock();
```

Comment:
This isn't going to work for the same reason. Because we cannot make `RequestContext` to extend `ReentrantLock`, how about creating `ReentrantLock` and storing to the `ctx.attr(...)` and using it?

Reply:
I'm having trouble with how to implement it. So I left a comment.🤔

Reply:
We may implement an atomic operation by comparing the returned value out of `ctx.setAttr(MAP, newMap)`.
For example:
```java
for (;;) {
  var oldMap = ...;
  ...
  if (ctx.setAttr(MAP, newMap)) == oldMap) {
     // `newMap` was successfully replaced with `oldMap`.
     break;
  }
}
```

Reply:
🔎Please check the CAS algorithm implementation carefully. I'm a little worried.